lib/tasks/schools_import.rake

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# To get the latest version of the zip:
+# 1. Go to https://get-information-schools.service.gov.uk/Downloads
+# 2. Check "Establishment fields CSV"
+# 3. Submit
+# 4. Download the zip file
+# 5. Place it in db/data/edubasealldata.zip
+namespace :schools do
+  desc "Import schools from DfE GIAS CSV"
+  task import: :environment do
+    require "csv"
+    require "zip"
+
+    zip_file = Rails.root.join("db/data/edubasealldata.zip")
+    puts "Starting schools import. Total locations: #{Location.count}"
+
+    Zip::File.open(zip_file) do |zip|
+      csv_entry = zip.glob("*.csv").first
+      csv_content = csv_entry.get_input_stream.read
+
+      total_rows = CSV.parse(csv_content).count - 1 # Subtract 1 for header
+      processed_rows = 0
+      progress_bar_width = 50
+      batch_size = 1000
+      locations = []
+
+      CSV.parse(
+        csv_content,
+        headers: true,
+        encoding: "ISO-8859-1:UTF-8"
+      ) do |row|
+        locations << Location.new(
+          urn: row["URN"],
+          name: row["EstablishmentName"],
+          address: [
+            row["Street"],
+            row["Locality"],
+            row["Address3"]
+          ].compact.join(", "),
+          town: row["Town"],
+          county: row["County (name)"],
+          postcode: row["Postcode"],
+          url: row["SchoolWebsite"]
+        )
+
+        if locations.size >= batch_size
+          Location.import locations,
+                          on_duplicate_key_update: {
+                            conflict_target: [:urn],
+                            columns: %i[name address town county postcode url]
+                          }
+          locations.clear
+        end
+
+        # Update progress bar
```

Comment:
`ruby-progressbar` is already in `Gemfile.lock`, what about using that? Also looks easy to use and has 1.6K ⭐s on GitHub!!!

https://github.com/jfelchner/ruby-progressbar